src/appservice/MjolnirManager.ts

```diff
@@ -0,0 +1,224 @@
+            });
+
+            const mjolnir = await this.makeInstance(requestingUserId, managementRoomId, mjolnirClient);
+            await mjolnir.createFirstList(requestingUserId, "list");
+
+            await this.dataStore.store({
+                local_part: mjolnirLocalPart,
+                owner: requestingUserId,
+                management_room: managementRoomId,
+            });
+
+            return [mjolnirUserId, managementRoomId];
+        } else {
+            throw new Error(`User: ${requestingUserId} has already provisioned ${provisionedMjolnirs.length} mjolnirs.`);
+        }
+    }
+
+    /**
+     * Utility that creates a matrix client for a virtual user on our homeserver with the specified loclapart.
+     * @param localPart The localpart of the virtual user we need a client for.
+     * @returns A tuple with the complete mxid of the virtual user and a MatrixClient.
+     */
+    private async makeMatrixClient(localPart: string): Promise<[string, MatrixClient]> {
+        // Now we need to make one of the transparent mjolnirs and add it to the monitor.
+        const mjIntent = await this.bridge.getIntentFromLocalpart(localPart);
+        await mjIntent.ensureRegistered();
+        return [mjIntent.userId, mjIntent.matrixClient];
+    }
+
+    // TODO: We need to check that an owner still has access to the appservice each time they send a command to the mjolnir or use the web api.
+    // https://github.com/matrix-org/mjolnir/issues/410
+    /**
+     * Used at startup to create all the ManagedMjolnir instances and start them so that they will respond to users.
+     */
+    private async createMjolnirsFromDataStore() {
+        for (const mjolnirRecord of await this.dataStore.list()) {
+            const [_mjolnirUserId, mjolnirClient] = await this.makeMatrixClient(mjolnirRecord.local_part);
+            const access = this.accessControl.getUserAccess(mjolnirRecord.owner);
+            if (access.outcome !== Access.Allowed) {
+                // Don't await, we don't want to clobber initialization just because we can't tell someone they're no longer allowed.
+                mjolnirClient.sendNotice(mjolnirRecord.management_room, `Your mjolnir has been disabled by the administrator: ${access.rule?.reason ?? "no reason supplied"}`);
+            } else {
+                await this.makeInstance(
+                    mjolnirRecord.owner,
+                    mjolnirRecord.management_room,
+                    mjolnirClient,
+                );
+            }
+        }
+    }
+}
+
+// Isolating this mjolnir is going to require provisioning an access token just for this user to be useful.
+// We can use fork and node's IPC to inform the process of matrix evnets.
+export class ManagedMjolnir {
+    public constructor(
+        public readonly ownerId: string,
+        private readonly mjolnir: Mjolnir,
+    ) { }
+
+    public async onEvent(request: Request<WeakEvent>) {
+        // phony sync.
```

Comment:
I *think* I understand what you mean here, but some clarification would be nice :)